src/main/src/main/java/org/geoserver/catalog/LayerGroupHelper.java

```diff
@@ -309,4 +427,79 @@ private static boolean isGroupInStack(LayerGroupInfo group, Stack<LayerGroupInfo
         
         return false;
     }
+
+    private static Catalog getCatalogFromStyle(StyleInfo info) {
+        info = ModificationProxy.unwrap(info);
+        if (info instanceof StyleInfoImpl) {
+            return ((StyleInfoImpl) info).getCatalog();
+        }
+        throw new IllegalStateException("Could not get catalog from style group");
+    }
+
+    private static void expandStyleGroup(StyleInfo styleGroup, CoordinateReferenceSystem crs, List<LayerInfo> layers, List<StyleInfo> styles) {
+        if (layers == null) {
+            layers = new ArrayList<>();
+        }
+        if (styles == null) {
+            styles = new ArrayList<>();
+        }
+
+        try {
+            Catalog catalog = getCatalogFromStyle(styleGroup);
+            StyledLayerDescriptor sld = ResourcePool.create(catalog).dataDir().parsedSld(styleGroup);
+            StyleGroupHelper helper = new StyleGroupHelper(catalog, crs);
+            sld.accept(helper);
+            layers.addAll(helper.getLayers());
+            styles.addAll(helper.getStyles());
+        } catch (IllegalStateException | IOException | ServiceException | UncheckedIOException | UnsupportedOperationException e) {
+            LOGGER.log(Level.WARNING, "Error extracting styles from Style Group '" + styleGroup.getName() + "'. Skipping...", e);
+        }
+    }
+
+    protected static class StyleGroupHelper extends GeoServerSLDVisitorAdapter {
```

Comment:
want a javadoc if you are going to make a protected class (or lock it down to private until you get a subclass)